From a4da7d7a9537b45a139f35733ea10ee3b46bc0ad Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:40:54 +0100
Subject: [PATCH 01/12] don't require rust

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f30ad57..b27aaa5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,6 @@ include(SetupBuildkite)
 include(SetupBun)
 include(SetupEsbuild)
 include(SetupZig)
-include(SetupRust)
 
 include(SetupCcache)
 
