From d99731a2f28a0a40cb7f294e29c2c547422fe083 Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:42:34 +0100
Subject: [PATCH 05/12] add targets to register_command

---
 cmake/targets/BuildBun.cmake | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index 39de33b..4eb32b2 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -1373,6 +1373,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1473,6 +1475,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1496,6 +1500,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
@@ -1552,6 +1558,8 @@ if(NOT BUN_CPP_ONLY)
     register_command(
       TARGET
         ${bun}
+      TARGETS
+        ${bun}
       TARGET_PHASE
         POST_BUILD
       COMMENT
