From 2894f949d00d565692210fa1d1c31ffdd750d595 Mon Sep 17 00:00:00 2001
From: Pavel Zwerschke <pavelzw@gmail.com>
Date: Sat, 7 Feb 2026 21:46:55 +0100
Subject: [PATCH 07/12] disable -ld_new

---
 cmake/targets/BuildBun.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index cce35da..d08f98a 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -1067,7 +1067,6 @@ endif()
 
 if(APPLE)
   target_link_options(${bun} PUBLIC
-    -Wl,-ld_new
     -Wl,-no_compact_unwind
     -Wl,-stack_size,0x1200000
     -fno-keep-static-consts
