From 35ea82a8dc49ad73d9ed693ea60d9f68d27d1e4f Mon Sep 17 00:00:00 2001
From: I-Al-Istannen <i-al-istannen@users.noreply.github.com>
Date: Tue, 10 Feb 2026 22:25:43 +0100
Subject: [PATCH 09/12] Use system ICU

---
 cmake/targets/BuildBun.cmake | 18 ++++--------------
 1 file changed, 4 insertions(+), 14 deletions(-)

diff --git a/cmake/targets/BuildBun.cmake b/cmake/targets/BuildBun.cmake
index d08f98a..c3060e5 100644
--- a/cmake/targets/BuildBun.cmake
+++ b/cmake/targets/BuildBun.cmake
@@ -1306,11 +1306,6 @@ target_link_libraries(${bun} PRIVATE
   zstd
 )
 
-if(APPLE)
-  target_link_libraries(${bun} PRIVATE icucore resolv)
-  target_compile_definitions(${bun} PRIVATE U_DISABLE_RENAMING=1)
-endif()
-
 if(USE_STATIC_SQLITE)
   target_compile_definitions(${bun} PRIVATE LAZY_LOAD_SQLITE=0)
 else()
@@ -1325,17 +1320,12 @@ if(LINUX)
   else()
     target_link_libraries(${bun} PUBLIC libatomic.so)
   endif()
-
-  if(WEBKIT_LOCAL)
-    find_package(ICU REQUIRED COMPONENTS data i18n uc)
-    target_link_libraries(${bun} PRIVATE ICU::data ICU::i18n ICU::uc)
-  else()
-    target_link_libraries(${bun} PRIVATE ${WEBKIT_LIB_PATH}/libicudata.a)
-    target_link_libraries(${bun} PRIVATE ${WEBKIT_LIB_PATH}/libicui18n.a)
-    target_link_libraries(${bun} PRIVATE ${WEBKIT_LIB_PATH}/libicuuc.a)
-  endif()
 endif()
 
+target_link_libraries(${bun} PRIVATE $ENV{PREFIX}/lib/libicudata$ENV{SHLIB_EXT})
+target_link_libraries(${bun} PRIVATE $ENV{PREFIX}/lib/libicui18n$ENV{SHLIB_EXT})
+target_link_libraries(${bun} PRIVATE $ENV{PREFIX}/lib/libicuuc$ENV{SHLIB_EXT})
+
 if(WIN32)
   target_link_libraries(${bun} PRIVATE
     winmm
